Packet++/header/DoIpLayer.h

```diff
@@ -0,0 +1,1947 @@
+
+		/// @brief Clears the optional max data size field.
+		void clearMaxDataSize();
+
+		/// @brief Returns a human-readable summary of the message.
+		/// @return A string summarizing the Entity Status Response.
+		std::string getSummary() const;
+
+		/// @brief  Checks if the Entity Status Response data length is valid.
+		/// @param[in] dataLen Length of the data buffer.
+		/// @return  true if the data length is valid, false otherwise.
+		static inline bool isDataLenValid(size_t dataLen)
+		{
+			return (dataLen == FIXED_LEN || dataLen == OPT_LEN);
+		}
+
+	private:
+#pragma pack(push, 1)
+		struct entity_status_response : doiphdr
+		{
+			uint8_t nodeType;
+
+			uint8_t maxConcurrentSockets;
+
+			uint8_t currentlyOpenSockets;
+		};
+#pragma pack(pop)
+
+		entity_status_response* getEntityStatusResponsePtr() const
+		{
+			return reinterpret_cast<entity_status_response*>(m_Data);
+		}
+		static constexpr size_t FIXED_LEN = sizeof(entity_status_response);
+		static constexpr size_t MAX_DATA_SIZE_OFFSET = FIXED_LEN;
+		static constexpr size_t MAX_DATA_SIZE_LEN = sizeof(uint32_t);
+		static constexpr size_t OPT_LEN = FIXED_LEN + MAX_DATA_SIZE_LEN;
+	};
+
+	//~~~~~~~~~~~~~~~~~~~|
+	// DoIpDiagnosticBase|
+	//~~~~~~~~~~~~~~~~~~~|
+
+	/// @class DoIpDiagnosticBase
+	/// @brief Represents a DoIP Diagnostic Message sent between tester and ECU.
+	/// This class includes source and target addresses.
+	class DoIpDiagnosticBase : public DoIpLayer
+	{
+	public:
+		/// @brief Constructs the DiagnosticMessage from raw packet data.
+		/// @param[in] data Pointer to the raw payload data.
+		/// @param[in] dataLen Length of the data buffer.
+		/// @param[in] prevLayer Pointer to the previous protocol layer.
+		/// @param[in] packet Pointer to the parent packet.
+		DoIpDiagnosticBase(uint8_t* data, size_t dataLen, Layer* prevLayer, Packet* packet);
+
+		/// @brief default c'tor.
+		DoIpDiagnosticBase(size_t length) : DoIpLayer(length) {};
+
+		/// @brief Gets the source logical address of the message.
+		/// @return 16-bit address of the source ECU.
+		uint16_t getSourceAddress() const;
+
+		/// @brief Gets the target logical address of the message.
+		/// @return 16-bit address of the destination ECU.
+		uint16_t getTargetAddress() const;
+
+		/// @brief Sets the source logical address.
+		/// @param[in] sourceAddress 16-bit source address.
+		void setSourceAddress(uint16_t sourceAddress);
+
+		/// @brief Sets the target logical address.
+		/// @param[in] targetAddress 16-bit target address.
+		void setTargetAddress(uint16_t targetAddress);
+
+		/// @brief Returns a human-readable summary of the message content.
+		/// @return A string summarizing the diagnostic message.
+		virtual std::string getSummary() const = 0;
+
+	protected:
+/// @struct  common_diagnostic_header
+/// Common first diagnostic data in DoIP
+/// messages (diagnostic/diagnosticAck/diagnosticNack).
+/// common_diagnostic_header message structure (extends DoIP header).
```

Comment:
nit: we can remove this comment and the comment on every field to stay consistent with the other protected structs